src/Network/DNS.hs

```diff
@@ -98,18 +98,22 @@ import           Compat
 import           Network.DNS.FFI
 import           Network.DNS.Message
 
--- | Exception thrown in case of errors while encoding or decoding into a 'Msg'.
+-- | Exception thrown in case of errors while resolving or encoding/decoding into a 'Msg'.
 --
 -- @since 0.1.1.0
 data DnsException = DnsEncodeException
                   | DnsDecodeException
+                  | DnsHostNotFound -- ^ No such domain (authoritative)
+                  | DnsNoData       -- ^ No record for requested type
+                  | DnsNoRecovery   -- ^ Non recoverable errors, REFUSED, NOTIMP
+                  | DnsTryAgain     -- ^ No such domain (non-authoritative) or SERVERFAIL
```

Comment:
It seems that it is not possible to combine `@since` and description for a single record field, see https://haskell-haddock.readthedocs.io/en/latest/markup.html#since (`@since` has to be followed with a version number, no further description is currently allowed).

Reply:
An option is to remove descriptions as fields names are self-descriptive.

Reply:
About the changelog. Should it contain other changes, e.g. the fix of memory leaks?